packages/components/organisms/f-loyalty/src/components/Header.vue

```diff
@@ -0,0 +1,139 @@
+                :router-links="true" />
+            <media-element
+                :class="$style['c-loyaltyHeader-media']"
+                :text-size="fontSize"
+                :title="$t('header.title')"
+                :text="$t('header.text')"
+                :image-url="getImageBasedOnScreenSize"
+                image-align="center"
+                :flex="flexLayout"
+                :alt="$t('header.alt')"
+            />
+        </div>
+    </div>
+</template>
+
+<script>
+import BreadCrumbs from '@justeat/f-breadcrumbs';
+import MediaElement from '@justeat/f-media-element';
+import '@justeat/f-media-element/dist/f-media-element.css';
+import '@justeat/f-breadcrumbs/dist/f-breadcrumbs.css';
+import debounce from 'lodash.debounce';
+import { FONT_SIZE } from '@justeat/f-media-element/src/constants';
+
+const REQUIRED_WINDOW_WIDTH = 600;
+const TEMP_IMG_PATH = 'https://d30v2pzvrfyzpo.cloudfront.net/b/hw/img/decoration/';
+
+export default {
+    name: 'StampcardsHeader',
+
+    components: {
+        BreadCrumbs,
+        MediaElement
+    },
+
+    data: () => ({
+        links:  [
+            {
+                name: 'Home',
+                url: '/'
+            },
+            {
+                name: 'For You',
+                url: '/offers'
+            },
+            {
+                name: 'StampCards',
+                url: '/offers/stamp-cards'
+            }
+        ],
+        fontSize: FONT_SIZE.LG,
+        windowWidth: REQUIRED_WINDOW_WIDTH
+    }),
+
+    computed: {
+        /**
+         * Returns the image based on the current window size
+         * @returns {string}
+         */
+        getImageBasedOnScreenSize () {
+            return this.windowWidth >= REQUIRED_WINDOW_WIDTH
+                ? `${TEMP_IMG_PATH}stampcards-header-${this.$t('percentage')}.svg`
+                : `${TEMP_IMG_PATH}stampcards-header-small.svg`;
+        },
+
+        /**
+         * Sets the media element font size based on the current screen size
+         * @returns {string}
+         */
+        getFontSizeBasedOnScreenSize () {
+            return this.windowWidth >= REQUIRED_WINDOW_WIDTH
+                ? 'xl' : 'md';
+        },
+
+        flexLayout () {
+            return {
+                default: {
+                    column: false,
+                    reverse: false
+                }
+            };
+        }
+    },
+
+    destroyed () {
+        if (process.browser) window.removeEventListener('resize', this.handleResize);
+    },
+
+    mounted () {
+        if (process.browser) window.addEventListener('resize', debounce(this.handleResize, 300));
+        this.handleResize();
+    },
+
+    methods: {
+        /**
+         * Handles the window resize callback
+         */
+        handleResize () {
+            this.windowWidth = process.browser && window.innerWidth;
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+
+$responsive-mobileViewBreakpoint: '<=narrowMid';
```

Comment:
Not sure if you're planning on using this elsewhere in future PRa, but while it's only used in one place atm, seems like a premature abstraction.

Reply:
agreed